move_backwards_recovery/include/move_backwards_recovery/move_backwards_recovery.h

```diff
@@ -48,6 +50,7 @@ namespace move_backwards_recovery{
       double maxRecoveriesResetDistance_;
       int inPlaceRecoveryCount_;
       int maxInPlaceRecoveries_;
+      bool canceled_;
```

Comment:
You should use std::atomic to avoid race conditions.  Your boolean is being written in one thread and read in a different thread.

Reply:
done